### PR TITLE
qnnpack quantized activations: fix memory format issues

### PR DESCRIPTION
**Pull request by vkuzo**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#46077 qnnpack quantized activations: fix memory format issues**

Summary:

Some of QNNPACK quantized kernels were not handling NHWC correctly,
the data written respected the input format but the memory flag
was always set to contiguous.  This led to incorrect outputs.  This PR
1. adds testing for NHWC for qnnpack activations
2. fixes those activations which did not set the memory format on the output so that the new tests pass

Test Plan:

```
python test/test_quantization.py TestQuantizedOps.test_qhardsigmoid
python test/test_quantization.py TestQuantizedOps.test_leaky_relu
python test/test_quantization.py TestQuantizedOps.test_hardswish
python test/test_quantization.py TestQNNPackOps.test_qnnpack_tanh
python test/test_quantization.py TestQNNPackOps.test_qnnpack_sigmoid
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24213257](https://our.internmc.facebook.com/intern/diff/D24213257)

### CODE REVIEW

File: test/quantization/test_quantized_op.py [hunk 1]
```diff
@@ -323,25 +337,30 @@ def test_leaky_relu_observed_output(self, X):
...
-        self.assertEqual(qY.dequantize(), qY_hat.dequantize(),
-                         msg="F.leaky_relu failed ({} vs {})".format(qY, qY_hat))
+            # torch.nn.functional
+            op = torch.nn.functional.leaky_relu
+            dqY = op(dqX, negative_slope=alpha)
+            qY = torch.quantize_per_tensor(dqY, scale=scale, zero_point=zero_point,
+                                           dtype=torch_type)
+            qY_hat = op(qX, negative_slope=alpha)
+            self.assertEqual(qY.dequantize(), qY_hat.dequantize(),
+                             msg="F.leaky_relu failed ({} vs {})".format(qY, qY_hat))
```
Comment (supriyar) [inline test/quantization/test_quantized_op.py:363]: For `leaky_relu` and the ones below, should we be using `_test_activation_function` to be consistent?

**Review by supriyar — APPROVED** [review]
Nice catch! This is a sneaky bug

File: test/quantization/test_quantized_op.py (hunk 1 above)
Comment (vkuzo) [reply test/quantization/test_quantized_op.py:363]: I think that would be valuable for the future.  We are hoping to land this urgently (FB teams blocked by this, and also to be able to pick into OSS v1.7), so maybe we can improve the test infra as a follow-up PR.  I would also recommend reducing our exposure to hypothesis in these kinds of tests, it seems to be a net negative in my experience so far, as the tensor values it gives are not representative of real inputs, which could mask issues.
